### ISSUE TITLE
Avoid exception handling for PostgresTarget.create_marker_table() with "CREATE TABLE IF NOT EXISTS"

### ISSUE THREAD

**Issue by ant1j** [spotify/luigi#2060]
I think it would be more elegant to add `IF NOT EXISTS` in the query for creating the `marker_table` in `luigi.contrib.postgres` module, instead of dealing with an exception to handle the case where the table already exists.

```sql
CREATE TABLE IF NOT EXISTS {marker_table} {
update_id TEXT PRIMARY KEY,
target_table TEXT,
inserted TIMESTAMP DEFAULT NOW())
```

`IF NOT EXISTS` is available since postgres 9.1 (released more than 5 years ago) and it seems that it is also available in many RDBMS.

So if the method should be rewrote I guess it would be as follows:

```python
    def create_marker_table(self):
        """
        Create marker table if it doesn't exist.

        Using a separate connection since the transaction might have to be reset.
        """
        connection = self.connect()
        connection.autocommit = True
        cursor = connection.cursor()
        if self.use_db_timestamps:
            sql = """ CREATE TABLE IF NOT EXISTS {marker_table} (
                      update_id TEXT PRIMARY KEY,
                      target_table TEXT,
                      inserted TIMESTAMP DEFAULT NOW())
                """.format(marker_table=self.marker_table)
        else:
            sql = """ CREATE TABLE IF NOT EXISTS {marker_table} (
                      update_id TEXT PRIMARY KEY,
                      target_table TEXT,
                      inserted TIMESTAMP);
                  """.format(marker_table=self.marker_table)
        try:
            cursor.execute(sql)
        except psycopg2.ProgrammingError as e:
                # in case there is another exception, i.e. other than a duplicated table name, raise
                raise
        connection.close()
```

Again I am not experienced enough to make a PR, so if any good soul can manage to do it...

Thanks

**Comment by Tarrasch** [comment]
> Again I am not experienced enough to make a PR, so if any good soul can manage to do it...

Not correct. Anyone can make a PR. All previous, current and future contributors are humans and typically make mistakes. We would be happy to accept your bugfix. :)

**Comment by ant1j** [comment]
Here you go.
You were true, anyone can make a PR!

**Comment by dlstadther** [comment]
resolved by #2064
